packages/by-name/kata/kata-runtime/0005-genpolicy-deny-UpdateEphemeralMountsRequest.patch

```diff
@@ -0,0 +1,75 @@
+From a72dc578428a828a8b03ae58a2c87a6565a2776b Mon Sep 17 00:00:00 2001
+From: Markus Rudy <mr@edgeless.systems>
+Date: Fri, 19 Jul 2024 11:08:19 +0200
+Subject: [PATCH 05/13] genpolicy: deny UpdateEphemeralMountsRequest
+
+The UpdateEphemeralMountsRequest is triggered by memory hotplug events,
+which are not supported for TEEs. Denying this request by default
+spares us the implementation of fine-grained target checks. The default
+can still be overridden with the settings file.
```

Comment:
That docstring is not wrong, but afaict you can _also_ mount all sorts of other things with this (e.g. NFS on `/contrast/tls-config`). So instead of checking every `storage` entry, we just disable it.

This is a cherry-pick of https://github.com/microsoft/kata-containers/commit/9a780633dbc1f75f3a5577eec0d06b0f3db39a8c, I should add that to the commit message.

Reply:
> That docstring is not wrong, but afaict you can _also_ mount all sorts of other things with this (e.g. NFS on `/contrast/tls-config`). So instead of checking every `storage` entry, we just disable it.

My concern was more with regards to the commits message. It talks about memory hotplug, how is that related to mounting directories?

Reply:
The argument for _why we don't need to allow this at all_ is that `updateEphemeralMounts` is _only_ called from the memory hotplug handler - I'll try to make that more clear in the message.

Reply:
(as to why memory hotplugging needs tmpfs remounts: maybe to adjust their size limits?)

Reply:
I'm now cherry-picking the upstream commit directly.